sherpa/astro/xspec/src/_xspec.cc

```diff
@@ -133,262 +62,118 @@ static int _sherpa_init_xspec_library()
     return EXIT_FAILURE;
   }
 
-  // Stream buffer for redirected stdout
-  std::streambuf* cout_sbuf = NULL;
-  std::ofstream fout;
+  // Redirect the stdout channel for the duration of the FNINIT call.
+  // The replacement is set to ostringstream so that we can find out
+  // what was sent to it in case of emergency (but would need code
+  // changes to do this). This used to be done by manually replacing
+  // the std::cout buffer.
+  //
+  // Note that only FNINIT needs this, as the FunctionUtility calls do
```

Comment:
Does this only need to be handled for FNINIT? I see the same stream handling applied elsewhere in this file (although for STDERR instead of STDOUT)

Reply:
I'll have to review, as it's been a while since I last looked at it. My guess is

1. I had hoped that we could avoid this entirely
2. but we can't, so we could wrap it neatly
3. but then you realize some other routine needs something similar and you don't update the existing documentation

Reply:
Hopefully the documentation for this routine is better. It's actually a "less is more" since much of the comments were describing how it has changed behaviour, but we don't really care about the old code now.